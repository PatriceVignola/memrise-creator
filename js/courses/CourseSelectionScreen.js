/* @flow */

import React from 'react';
import { FlatList } from 'react-native';
import { connect } from 'react-redux';
import fetchCoursesAsync from '../actions/fetchApi';
import type { Course } from '../reducers/courses';
import type { Action } from '../actions/types';
import CourseCard from './CourseCard';

type Props = {
  courses: Course[],
  dispatch: ((Action => void) => Promise<Action>) => void,
};

type State = {
  courses: Course[],
};

// Name-export the unconnected component for tests
export class CourseSelectionScreen extends React.Component<Props> {
  componentDidMount() {
    console.warn('did mount');
    this.props.dispatch(fetchCoursesAsync());
  }

  render() {
    return (
      // TODO: Put loading gif while the courses are loading
      <FlatList
        data={this.props.courses}
        renderItem={(info: { item: Course }) => (
          <CourseCard course={info.item} />
        )}
      />
    );
  }
}

function mapStateToProps(state: State) {
  return {
    courses: state.courses,
  };
}

export default connect(mapStateToProps)(CourseSelectionScreen);
