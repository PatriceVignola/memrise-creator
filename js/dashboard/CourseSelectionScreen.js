/* @flow */

import React from 'react';
import {
  Button,
  Image,
  StyleSheet,
  Text,
  View,
  WebView,
} from 'react-native';
import getCoursesAsync from '../api/courseApi';

type State = {
  // TODO: Strongly type courses with the type from courseApi
  courses: Array<Object>,
  loadingCourses: boolean,
  courseUrl: string,
};

export default class CourseSelectionScreen extends React.Component<{}, State> {
  webView: WebView;

  constructor() {
    super();

    this.state = {
      // TODO: See if we can make it strongly typed with Flow
      courses: [],
      loadingCourses: true,
      courseUrl: 'https://www.memrise.com/course/1908937/test-korean-3/edit/#l_7094047',
    };

    this.webView = null;

    (this: any).onAddPress = this.onAddPress.bind(this);
  }

  componentDidMount() {
    console.warn('did mount');
    getCoursesAsync().then((courses) => {
      console.warn(courses);
      this.setState({ loadingCourses: false, courses });
    });
  }

  onAddPress() {
    console.warn('onAddPress()');
    this.webView.injectJavaScript('$("input.wide")[0].value = "LALALA"; $("input.wide")[1].value = "LOLLOL"; $(".ico-blue").click();');
  }

  render() {
    return (
      // TODO: Put loading gif
      <View>
        {this.state.loadingCourses &&
          <Text>Loading courses...</Text>
        }

        <Button title="ADD!" onPress={this.onAddPress} />
        <WebView
          ref={(ref) => { this.webView = ref; }}
          source={{ uri: this.state.courseUrl }}
          userAgent="Mozilla/5.0 (compatible; MSIE 10.0; Windows Phone 8.0; Trident/6.0; IEMobile/10.0; ARM; Touch; NOKIA; Lumia 920)"
        />

        {this.state.courses.map(course => (
        // TODO: Make a "Course" component to group all this
        // TODO: Use a google material cards view to show them
          <View key={course.id}>
            <Image style={styles.courseImage} source={{ uri: course.photo }} />
            <Image style={styles.courseImage} source={{ uri: course.photo_small }} />
            <Image style={styles.courseImage} source={{ uri: course.photo_large }} />
            <Text>ID: {course.id}</Text>
            <Text>Name: {course.name}</Text>
            <Text>Slug: {course.slug}</Text>
            <Text>URL: {course.url}</Text>
            <Text>Description: {course.description}</Text>
            <Text>Num Things: {course.num_things}</Text>
            <Text>Num Levels: {course.num_levels}</Text>
          </View>
        ))}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  courseImage: {
    width: 50,
    height: 50,
  },
});
