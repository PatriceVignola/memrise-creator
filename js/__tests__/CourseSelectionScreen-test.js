/* @flow */

import React from 'react';
import renderer from 'react-test-renderer';
import CourseSelectionScreen from '../dashboard/CourseSelectionScreen';

test('renders correctly', () => {
  const tree = renderer.create(<CourseSelectionScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
