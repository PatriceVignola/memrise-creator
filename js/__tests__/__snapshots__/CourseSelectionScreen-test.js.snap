// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Loading courses...
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        ADD!
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={undefined}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={undefined}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={true}
      scrollEnabled={undefined}
      source={
        Object {
          "uri": "https://www.memrise.com/course/1908937/test-korean-3/edit/#l_7094047",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
